DOC-6747: Docs for Improve network observability inside CockroachDB (#17824)
cockroachdb/docs: src/current/v23.2/ui-network-latency-page.md

---
title: Network Page
summary: The Network Latency page displays round-trip latencies between all nodes in your cluster.
toc: true
---

> The **Network** page displays round-trip latencies between all nodes in your cluster. Latency is the time required to transmit a packet across a network, and is highly dependent on your network topology. Use this page to determine whether your latency is appropriate for your [topology pattern]({% link {{ page.version.version }}/topology-patterns.md %}), or to identify nodes with unexpected latencies or connectivity issues.

Author: kept "between" because it is used in [this context](https://www.grammarly.com/blog/between-among/#:~:text=In%20fact%2C%20you%20can%20use%20between%20for%20any%20number%20of%20elements%2C%20as%20long%20as%20all%20the%20elements%20are%20separate%20and%20distinct.%20According%20to%20The%20Chicago%20Manual%20of%20Style%2C%20you%20can%20even%20use%20between%20when%20%E2%80%9Cmultiple%20one%2Dto%2Done%20relationships%20are%20understood%20from%20the%20context%E2%80%9D%3A).

updated as suggested: "This represents" and "and".

> The **Network** page displays round-trip latencies between all nodes in your cluster. This represents the time required to transmit a packet across a network, and is highly dependent on your network topology. Use this page to determine whether your latency is appropriate for your [topology pattern]({% link {{ page.version.version }}/topology-patterns.md %}), and to identify nodes with unexpected latencies or connectivity issues.

Reviewer: ```suggestion
The **Network** page displays round-trip latencies between all nodes in your cluster. This represents the time required to transmit a packet across a network, and is highly dependent on your network topology. Use this page to determine whether your latency is appropriate for your [topology pattern]({% link {{ page.version.version }}/topology-patterns.md %}), and to identify nodes with unexpected latencies or connectivity issues such as a [network partition](#network-partition).
```

To view this page, [access the DB Console]({% link {{ page.version.version }}/ui-overview.md %}#db-console-access) and click **Network** in the left-hand navigation.

> ## Sort and filter network matrix

Author: updated with "latencies"

Use the **Sort By** menu to arrange the network matrix by [locality]({% link {{ page.version.version }}/cockroach-start.md %}#locality) (e.g., cloud, region, availability zone, datacenter).

Use the **Filter** menu to select specific nodes or localities to view.

Select **Collapse Nodes** to display the mean latencies of each locality, depending on how the matrix is sorted. This is a way to quickly assess cross-regional or cross-cloud latency.

> ## Understanding the network matrix

Author: updated

> Each cell in the matrix displays the round-trip latency in milliseconds between two nodes in your cluster. Round-trip latency includes the return time of a packet. Latencies are color-coded by their standard deviation from the mean latency on the network: green for lower values, and blue for higher.

Author: updated

<img src="{{ 'images/v23.2/ui_network_latency_matrix.png' | relative_url }}" alt="DB Console Network Latency matrix" style="border:1px solid #eee;max-width:100%" />

> Rows represent origin nodes, and columns represent destination nodes. Hover over a cell to see:

Author: updated

> - the direction of the connection marked by `From` and `To`

Author: updated

- locality metadata for origin and destination
- round-trip latency 

> The page auto-refreshes every 30 seconds to show the most recent information about the nodes and their network connections.

Author: updated

### No connections

> Nodes that have lost a connection are color-coded depending on connection status: orange with an exclamation mark for `unknown connection state`, and red with a [no symbol](https://en.wikipedia.org/wiki/No_symbol) for `Failed connection`. This can help you locate a network partition in your cluster.

Author: added screenshot.

This specific information can help you understand the root cause of the connectivity issue.

{{site.data.alerts.callout_info}}
A network partition prevents nodes from communicating with each other in one or both directions. This can be due to a configuration problem with the network, such as when allowlisted IP addresses or hostnames change after a node is torn down and rebuilt. In a symmetric partition, node communication is broken in both directions. In an asymmetric partition, node communication works in one direction but not the other.

The effect of a network partition depends on which nodes are partitioned, where the ranges are located, and to a large extent, whether [localities]({% link {{ page.version.version }}/cockroach-start.md %}#locality) are defined. If localities are not defined, a partition that cuts off at least (n-1)/2 nodes will cause data unavailability.
{{site.data.alerts.end}}

### Node liveness status

Hover over Node ID’s in the row and column headers to show node liveness status, such as `healthy` or `suspect`. Node liveness status is also indicated by the colored circle next to the Node ID: green for `healthy` or red for `suspect`.

> If a `suspect` node stays offline for the duration set by[`server.time_until_store_dead`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-time-until-store-dead) (5 minutes by default), the [cluster considers the node "dead"]({% link {{ page.version.version }}/node-shutdown.md %}#process-termination) and the node is removed from the matrix.

Author: updated.

### No connections

> Nodes that have lost a connection are color-coded depending on connection status: orange with an exclamation mark for `unknown connection state`, and red with a [no symbol](https://www.wikipedia.org/wiki/No_symbol) for `Failed connection`. This can help you locate a network partition in your cluster.

Reviewer: ```suggestion
Nodes that have completelyt lost connectivity are color-coded depending on connection status:

- Orange with an exclamation mark indicates `unknown connection state`.
- Red with a [no symbol](https://www.wikipedia.org/wiki/No_symbol) indicates `Failed connection`. 

This information can help you diagnose a network partition in your cluster.
```

Author: updated

<img src="{{ 'images/v23.2/ui_network_latency_matrix_suspect_node.png' | relative_url }}" alt="DB Console Network Latency suspect node" style="border:1px solid #eee;max-width:100%" />

 Hover over a cell to see:

> - the direction of the connection marked by `From` and `To`

Reviewer: These should also be formatted as sentences.

Author: updated

- locality metadata for origin and destination
- connection status
- the last error message of the connection attempt

This specific information can help you understand the root cause of the connectivity issue.

{{site.data.alerts.callout_info}}
A network partition prevents nodes from communicating with each other in one or both directions. This can be due to a configuration problem with the network, such as when allowlisted IP addresses or hostnames change after a node is torn down and rebuilt. In a symmetric partition, node communication is broken in both directions. In an asymmetric partition, node communication works in one direction but not the other.

The effect of a network partition depends on which nodes are partitioned, where the ranges are located, and to a large extent, whether [localities]({% link {{ page.version.version }}/cockroach-start.md %}#locality) are defined. If localities are not defined, a partition that cuts off at least (n-1)/2 nodes will cause data unavailability.
{{site.data.alerts.end}}

### Node liveness status

> Hover over Node ID’s in the row and column headers to show node liveness status, such as `healthy` or `suspect`. Node liveness status is also indicated by the colored circle next to the Node ID: green for `healthy` or red for `suspect`.

Reviewer: ```suggestion
Hover over a node's ID in the row and column headers to show the node's liveness status, such as `healthy` or `suspect`. Node liveness status is also indicated by the colored circle next to the Node ID: green for `healthy` or red for `suspect`.
```

Author: updated

If a `suspect` node stays offline for the duration set by[`server.time_until_store_dead`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-time-until-store-dead) (5 minutes by default), the [cluster considers the node "dead"]({% link {{ page.version.version }}/node-shutdown.md %}#process-termination) and the node is removed from the matrix.

> To see the number of `LIVE` (healthy), `SUSPECT`, `DRAINING` and `DEAD` nodes, navigate to the [Cluster Overview page]({% link {{ page.version.version }}/ui-cluster-overview-page.md %}).

Reviewer: ```suggestion
The number of `LIVE` (healthy), `SUSPECT`, `DRAINING` and `DEAD` nodes is displayed in the [Cluster Overview page]({% link {{ page.version.version }}/ui-cluster-overview-page.md %}).
```

Author: updated

### No connections

Nodes that have completely lost connectivity are color-coded depending on connection status:

- Orange with an exclamation mark indicates `unknown connection state`.
- Red with a [no symbol](https://www.wikipedia.org/wiki/No_symbol) indicates `Failed connection`. 

This information can help you diagnose a network partition in your cluster.

<img src="{{ 'images/v23.2/ui_network_latency_matrix_suspect_node.png' | relative_url }}" alt="DB Console Network Latency suspect node" style="border:1px solid #eee;max-width:100%" />

Hover over a cell to display more details:

- The direction of the connection marked by `From` and `To`.
- Locality metadata for origin and destination.
- Connection status.
- The error message that resulted from the most recent connection attempt.

This specific information can help you understand the root cause of the connectivity issue.

> {{site.data.alerts.callout_info}}

Reviewer: It seems like this information can be moved out of the admonition and de-duplicated with the info above. I think you have defined a network partition pretty well by this point in the content. Sorry I didn't notice this before.

Author: I don't see how to "de-duplicate" the call out. So I am leaving it as is.

Reviewer: Apologies for the confusion. I think I was thinking it would be useful to define "network partition" earlier, but now I see that it might be better to use the words early (for SEO), but to move the admonition text into a sub-section. Feel free to ignore these suggestions if you prefer the way it is now. :)

> {{site.data.alerts.callout_info}}

Reviewer: ```suggestion
### Network partitions

```

A network partition prevents nodes from communicating with each other in one or both directions. This can be due to a configuration problem with the network, such as when allowlisted IP addresses or hostnames change after a node is torn down and rebuilt. In a symmetric partition, node communication is broken in both directions. In an asymmetric partition, node communication works in one direction but not the other.

The effect of a network partition depends on which nodes are partitioned, where the ranges are located, and to a large extent, whether [localities]({% link {{ page.version.version }}/cockroach-start.md %}#locality) are defined. If localities are not defined, a partition that cuts off at least (n-1)/2 nodes will cause data unavailability.

> {{site.data.alerts.end}}

Reviewer: ```suggestion
```

### Node liveness status

Hover over a node's ID in the row and column headers to show the node's liveness status, such as `healthy` or `suspect`. Node liveness status is also indicated by the colored circle next to the Node ID: green for `healthy` or red for `suspect`.

> If a `suspect` node stays offline for the duration set by[`server.time_until_store_dead`]({% link {{ page.version.version }}/cluster-settings.md %}#setting-server-time-until-store-dead) (5 minutes by default), the [cluster considers the node "dead"]({% link {{ page.version.version }}/node-shutdown.md %}#process-termination) and the node is removed from the matrix.

Author: updated

> The number of `LIVE` (healthy), `SUSPECT`, `DRAINING` and `DEAD` nodes is displayed in the [Cluster Overview page]({% link {{ page.version.version }}/ui-cluster-overview-page.md %}).

Reviewer: Also decommissioning? Did I see something about that somewhere?

Author: Decommissioning is displayed in the Status column of the Node List on the Cluster Overview page, under Node Status on the top panel it is not displayed:
<img width="328" alt="Screenshot 2023-09-14 at 10 11 46 AM" src="https://github.com/cockroachdb/docs/assets/58752716/1f1ace2d-db49-40ee-bbd1-4133a32cb83d">

I qualified the sentence with "under Node Status".
